mod_papers/helper.php

```diff
@@ -13,150 +14,257 @@
-                    $bibtex = $work['work-citation']['citation'];
-                    $volume = '';
-                    $pages  = '';
-                    if (preg_match('/volume\\s?=\\s?{(\\d+)}/', $bibtex, $match)) {
-                        $volume = $match[1];
-                    }
-                    if (preg_match('/pages\\s?=\\s?{([0-9-]+)}/', $bibtex, $match)) {
-                        $pages = $match[1];
-                    }
-                }
-
-                if (!is_null($work['work-contributors'])) {
-                    foreach ($work['work-contributors']['contributor'] as $authors) {
-                        if (($authors === reset($work['work-contributors']['contributor'])) && ($authors === end($work['work-contributors']['contributor']))) {
-                            $output .= $authors['credit-name']['value'] . '<br>';
-                        } elseif ($authors === reset($work['work-contributors']['contributor'])) {
-                            $output .= $authors['credit-name']['value'];
-                        } elseif ($authors === end($work['work-contributors']['contributor'])) {
-                            $output .= ' and ' . $authors['credit-name']['value'] . '<br>';
-                        } else {
-                            $output .= ', ' . $authors['credit-name']['value'];
-                        }
-                    }
-                } else {
-                    //Get authorlist from bibtex
-                    if (preg_match('/author\\s?=\\s?{(.+)}/', $bibtex, $match)) {
-                        $authorstr = $match[1];
-                        $authors   = explode(" and ", $authorstr);
-                        foreach ($authors as $author) {
-                            if (($author === reset($authors)) && ($author === end($authors))) {
-                                $output .= $author . '<br>';
-                            } elseif ($author === reset($authors)) {
-                                $output .= $author;
-                            } elseif ($author === end($authors)) {
-                                $output .= ' and ' . $author . '<br>';
-                            } else {
-                                $output .= ', ' . $author;
-                            }
-                        }
-                    }
-                }
-
-                $output .= '<a href="http://dx.doi.org/' . $work['udoi'] . '">' . $work['journal-title']['value'] . ' <b>' . $volume . '</b> ' . $pages . ' (' . $work['publication-date']['year']['value'] . ')</a><br style="line-height:2.5em;">';
-            }
-        }
-
-        return $output;
-    }
+	/**
+	 * Retrieves the required informaton from ORCID
+	 *
+	 * @param   array  $orcids An object containing the ORCIDs to be searched
+	 *
+	 * @access public
+	 */
+
+	public static function getPapers($orcids){
+		$lastRunLog =  dirname(__FILE__) . '/lastrun.log';
```

Comment:
Good to see the caching done&mdash;my implementation was a slap dash MWE, never really intended it to be operating so long in production without this kind of addition.

Reply:
sometimes users update their page too often.